typegate/tests/runtimes/prisma/schema_generation.py

```diff
@@ -1,401 +1,497 @@
-            "children": db.link(t.array(g13("TreeNode")), field="parent"),
-            "parent": db.link(g13("TreeNode"), field="children"),
-        }
-    ).named("TreeNode")
+            "id": t.integer(as_id=True, config={"auto": True}),
+            "parent": db.link(t.ref("TreeNode"), field="children"),
+            "children": db.link(t.array(t.ref("TreeNode")), field="parent"),
+        },
+        name="TreeNode",
+    )
 
-    g13.expose(
+    g.expose(
         createTreeNode=db.create(tree_node),
     )
 
 
-with TypeGraph("one-to-one-self") as g14:
+# @typegraph()
+# def one_to_one_self(g: Graph):
+#     db = PrismaRuntime("test", "POSTGRES")
+#
+#     tree_node = t.struct(
+#         {
+#             "id": t.integer(as_id=True, config={"auto": True}),
+#             "parent": t.ref("TreeNode"),
+#             "children": t.array(t.ref("TreeNode")),
+#         },
+#         name="TreeNode",
+#     )
+#
+#     g.expose(
+#         createTreeNode=db.create(tree_node),
+#     )
+#
+
+
+@typegraph()
+def one_to_one_self(g: Graph):
     db = PrismaRuntime("test", "POSTGRES")
 
     list_node = t.struct(
         {
-            "id": t.uuid().as_id.config("auto"),
-            "next": g14("ListNode").optional().config("unique"),
-            "prev": g14("ListNode").optional(),
-        }
-    ).named("ListNode")
+            "id": t.uuid(as_id=True, config={"auto": True}),
+            "next": t.ref("ListNode").optional(config={"unique": True}),
+            "prev": t.ref("ListNode").optional(),
+        },
+        name="ListNode",
+    )
 
-    g14.expose(
+    g.expose(
         createListNode=db.create(list_node),
     )
 
-with TypeGraph("one-to-one-self-2") as g15:
+
+# @typegraph()
+# def one_to_one_self_2(g: Graph):
+#     db = PrismaRuntime("test", "POSTGRES")
+#
+#     tree_node = t.struct(
+#         {
+#             "id": t.integer(as_id=True, config={"auto": True}),
+#             "parent": db.link("TreeNode", field="children"),
+#             "children": db.link(t.array(t.ref("TreeNode")), field="parent"),
+#         },
+#         name="TreeNode",
+#     )
+#
+#     g.expose(
+#         createTreeNode=db.create(tree_node),
+#     )
+#
+
+
+@typegraph()
+def one_to_one_self_2(g: Graph):
     db = PrismaRuntime("test", "POSTGRES")
 
     list_node = t.struct(
         {
-            "id": t.uuid().as_id.config("auto"),
-            "prev": g15("ListNode").optional(),
-            "next": g15("ListNode").optional().config("unique"),
-        }
-    ).named("ListNode")
+            "id": t.uuid(as_id=True, config={"auto": True}),
+            "prev": t.ref("ListNode").optional(),
+            "next": t.ref("ListNode").optional(config={"unique": True}),
+        },
+        name="ListNode",
+    )
 
-    g15.expose(
+    g.expose(
         createListNode=db.create(list_node),
     )
 
 
 # TODO ambiguous targets
-# with TypeGraph("multiple-relationships") as g16:
+# @typegraph()
```

Comment:
A few commented cases in this file? What is the status?

Reply:
They are already solved but forgot to remove them.